query-service-impl/src/main/java/org/hypertrace/core/query/service/trino/converters/DefaultColumnRequestConverter.java

```diff
@@ -243,37 +243,60 @@ private void handleConversionForArrayColumnExpression(
 
     if (handleConversionForNullOrEmptyArrayLiteral(lhs, operator, builder, value)) return;
 
-    // support only equals and IN operator
-    // both of them are handled as contains check to align with existing implementation
-    if (operator.equals("=")
-        || operator.equals("IN")
-        || operator.equals("!=")
-        || operator.equals("NOT IN")) {
-      // add NOT operator to negate the match condition
-      if (operator.equals("!=") || operator.equals("NOT IN")) {
+    // Support only equals and IN operator
+    if (operator.equals("=") || operator.equals("!=")) {
+      // Equals (=): CONTAINS(ip_types, 'Bot')
+      // Not equals (!=): NOT CONTAINS(ip_types, 'Bot')
+      if (operator.equals("!=")) {
         builder.append("NOT ");
       }
+      builder.append("CONTAINS(");
       builder.append(lhs);
-      // overlap operator for array
-      builder.append(" && ");
+      builder.append(", ");
+      if (value.getValueType() == ValueType.STRING) {
+        builder.append(QUESTION_MARK);
+        paramsBuilder.addStringParam(value.getString());
+      } else {
+        throw new IllegalArgumentException(
+            String.format("Unsupported value {%s} for array column", value));
+      }
+      builder.append(")");
+    } else if (operator.equals("IN") || operator.equals("NOT IN")) {
+      // IN: CARDINALITY(ARRAY_INTERSECT(ip_types, ARRAY['Public Proxy', 'Bot'])) > 0
+      // NOT IN: CARDINALITY(ARRAY_INTERSECT(ip_types, ARRAY['Public Proxy', 'Bot'])) = 0
+      builder.append("CARDINALITY(");
```

Comment:
Nit: changes looks fine. For readability, would it make sense to move to a function?